client/cpp/encoder.cc

```diff
@@ -256,4 +331,89 @@ bool Encoder::EncodeString(const std::string& value, Bits* irr_out) const {
   return _EncodeStringInternal(value, &unused_bloom, &unused_prr, irr_out);
 }
 
+static uint8_t shifted(const Bits& bits, const int& index) {
+  // For an array of bytes, select the appopriate byte from a 4-byte
+  // integer value.
```

Comment:
It is important to document that bytes are enumerated in the big-endian order: index = 0 selects the most significant byte, index = 3 the least significant.